src/connection/WalletConnectV2.ts

```diff
@@ -0,0 +1,50 @@
+import { sendAnalyticsEvent } from '@uniswap/analytics'
+import { WalletConnect, WalletConnectConstructorArgs } from '@web3-react/walletconnect-v2'
+import { SupportedChainId } from 'constants/chains'
+import { Z_INDEX } from 'theme/zIndex'
+
+import { RPC_URLS } from '../constants/networks'
+
+// Avoid testing for the best URL by only passing a single URL per chain.
+// Otherwise, WC will not initialize until all URLs have been tested (see getBestUrl in web3-react).
```

Comment:
was this a web3-react or wc decision? this seems not ideal to me (cc @grabbou)

Reply:
`Promise.race()` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/race

?